collectors/cgroups.plugin/cgroup-name.sh.in

```diff
@@ -53,18 +53,26 @@ function docker_get_name_classic() {
 }
 
 function docker_get_name_api() {
-	local id="${1}"
-	if [ ! -S "${DOCKER_HOST}" ]; then
-		warning "Can't find ${DOCKER_HOST}"
+	local path="/containers/${1}/json"
+	if [ -z "${DOCKER_HOST}" ]; then
+		warning "No DOCKER_HOST is set"
 		return 1
 	fi
 	if ! command -v jq >/dev/null 2>&1; then
 		warning "Can't find jq command line tool. jq is required for netdata to retrieve docker container name using ${DOCKER_HOST} API, falling back to docker ps"
 		return 1
 	fi
-
-	info "Running API command: /containers/${id}/json"
-	JSON=$(echo -e "GET /containers/${id}/json HTTP/1.0\\r\\n" | nc -U "${DOCKER_HOST}" | grep '^{.*')
+    if [ -S "${DOCKER_HOST}" ]; then
+        info "Running API command: curl --unix-socket ${DOCKER_HOST} http://localhost${path}"
+        JSON=$(curl --unix-socket ${DOCKER_HOST} http://localhost${path})
+    elif [ "${DOCKER_HOST}" == "/var/run/docker.sock" ]; then
+        warning "Docker socket was not found at ${DOCKER_HOST}"
+        return 1
+    else
+        info "Running API command: curl ${DOCKER_HOST}${path}"
```

Comment:
In here, before we run the API, you can do the check:

if [ -f ${DOCKER_SOCKET} ]; then
      echo "Found ${DOCKER_SOCKET} as a file, i don't know what to do"
      return 1
else
      API CALL CODE HERE
fi

Reply:
This is definitely another case we could catch, but I feel that it's probably more likely to be an accidental directory rather than a file. If someone provides a Docker volume to a path that does not exist, I believe it ends up getting created as a directory.

You think I should add that?